src/urg_node_driver.cpp

```diff
@@ -139,204 +155,308 @@ bool UrgNode::updateStatus()
+        } else {
+          string_result << "The parameter " << parameter.get_name() <<
+            " should be between 1 and 99.\n";
+          result.successful = false;
+        }
+      } else {
+        string_result << "The parameter " << parameter.get_name() <<
+          " is of the wrong type, should be an integer.\n";
+        result.successful = false;
+      }
 
-  urg_node::URGConfig min, max;
-  srv_->getConfigMin(min);
-  srv_->getConfigMax(max);
+    } else if (parameter.get_name().compare("skip") == 0) {
+      if (parameter_type == rclcpp::ParameterType::PARAMETER_INTEGER) {
+        if (parameter.as_int() >= 0 && parameter.as_int() <= 9) {
+          result.successful &= true;
+        } else {
+          string_result << "The parameter " << parameter.get_name() <<
+            " should be between 0 and 9.\n";
+          result.successful = false;
+        }
+      } else {
+        string_result << "The parameter " << parameter.get_name() <<
+          " is of the wrong type, should be an integer.\n";
+        result.successful = false;
+      }
 
-  min.angle_min = urg_->getAngleMinLimit();
-  min.angle_max = min.angle_min;
-  max.angle_max = urg_->getAngleMaxLimit();
-  max.angle_min = max.angle_max;
+    } else {
+      string_result << "The parameter " << parameter.get_name() <<
+        " is not part of the reconfigurable parameters.\n";
+      result.successful = false;
+    }
+  }
+  result.reason = string_result.str();
 
-  srv_->setConfigMin(min);
-  srv_->setConfigMax(max);
+  return result;
 }
 
 void UrgNode::calibrate_time_offset()
 {
-  boost::mutex::scoped_lock lock(lidar_mutex_);
-  if (!urg_)
-  {
-    ROS_DEBUG_THROTTLE(10, "Unable to calibrate time offset. Not Ready.");
+  std::unique_lock<std::mutex> lock(lidar_mutex_);
+  if (!urg_) {
+    RCLCPP_DEBUG(this->get_logger(), "Unable to calibrate time offset. Not Ready.");
     return;
   }
-  try
-  {
+  try {
     // Don't let outside interruption effect lidar offset.
-    ROS_INFO("Starting calibration. This will take a few seconds.");
-    ROS_WARN("Time calibration is still experimental.");
-    ros::Duration latency = urg_->computeLatency(10);
-    ROS_INFO("Calibration finished. Latency is: %.4f.", latency.toSec());
-  }
-  catch (std::runtime_error& e)
-  {
-    ROS_FATAL("Could not calibrate time offset: %s", e.what());
-    ros::Duration(1.0).sleep();
-    ros::shutdown();
+    RCLCPP_INFO(this->get_logger(), "Starting calibration. This will take a few seconds.");
+    RCLCPP_WARN(this->get_logger(), "Time calibration is still experimental.");
+    rclcpp::Duration latency = urg_->computeLatency(10);
+    RCLCPP_INFO(this->get_logger(), "Calibration finished. Latency is: %.4f sec.",
+      (double)(latency.nanoseconds() * 1e-9));
+  } catch (std::runtime_error & e) {
+    RCLCPP_FATAL(this->get_logger(), "Could not calibrate time offset: %s", e.what());
+    rclcpp::sleep_for(std::chrono::seconds(1));
+    rclcpp::shutdown();
```

Comment:
This is another case that feels like a layering violation, and we might just want to re-raise the error.

Reply:
done in 32fef1a